client/src/components/annotators/VideoAnnotator.vue

```diff
@@ -1,121 +1,117 @@
-<script>
-import Vue from 'vue';
-import annotator from './annotator';
+<script lang="ts">
+import { defineComponent, PropType } from '@vue/composition-api';
+import useMediaController from './useMediaController';
 
-export default Vue.extend({
+export default defineComponent({
   name: 'VideoAnnotator',
 
-  mixins: [annotator],
-
   props: {
     videoUrl: {
       type: String,
       required: true,
     },
     videoPlayerAttributes: {
-      type: Object,
+      type: Object as PropType<{ [key: string]: string }>,
       default: () => ({}),
     },
-  },
-
-  computed: {
-    isLastFrame() {
-      return this.playing;
+    frameRate: {
+      type: Number,
+      required: true,
     },
   },
 
-  created() {
-    const video = document.createElement('video');
-    this.video = video;
-    video.preload = 'auto';
-    video.src = this.videoUrl;
-    Object.entries(this.videoPlayerAttributes).forEach(([key, val]) => {
-      this.video[key] = val;
-    });
+  setup(props, { emit }) {
+    const common = useMediaController({ emit });
```

Comment:
minor thing, should we have something a bit more descriptive than common?  I know that it is common between the Video/Image annotators.  commonMedia?  Or may separating out initially the geoJS map stuff from the actual media controls stuff?